pkg/cloud/services/compute/instance.go

```diff
@@ -486,6 +497,62 @@ func isDuplicate(list []string, name string) bool {
 	return false
 }
 
+func getOrCreatePort(is *Service, name string, portOpts infrav1.PortOpts) (*ports.Port, error) {
+	existingPorts, err := listPorts(is, ports.ListOpts{
```

Comment:
IMO these infra re-use logic blocks are problematic:
- what happens if the port you find with the same name does not meet the other requirements specified by the user?
- what happens if a port with this name exists in 2 different networks/subnets?
- what happens if another service takes this port before we call instance create?

Feel free to ignore this, since I realize that this behavior is consistent across the api. I may file a bug about this to cover all the parts of the API that do this.

Reply:
> this behavior is consistent across the api
this is where I was coming from and just matching based on `name` is "good enough"™ for us
also didn't want to complicate the PR too much

Reply:
Don't worry about it, I will open an issue for it